viewer/src/component/three/RevealManager.ts

```diff
@@ -0,0 +1,291 @@
+                return loadCadModelFromCdf(client, model.source.modelRevision);
+              } else if (model.source.discriminator === 'external') {
+                return loadCadModelByUrl(model.source.url);
+              } else {
+                throw new Error('Unknown Source');
+              }
+            },
+            (model, cadModel) => ({
+              ...model,
+              data: cadModel
+            })
+          )
+        );
+        const pointCloudObservable = addModelObservable.pipe(
+          filter((model): model is PointCloud => model.format === File3dFormat.EptPointCloud),
+          flatMap(
+            model => {
+              if (model.source.discriminator === 'cdf') {
+                return createPointCloudModel(client, model.source.modelRevision);
+              } else if (model.source.discriminator === 'external') {
+                return createLocalPointCloudModel(model.source.url);
+              } else {
+                throw new Error('Unknown Source');
+              }
+            },
+            (model, pointCloudModel) => ({
+              ...model,
+              data: pointCloudModel
+            })
+          )
+        );
+        return merge(
+          cadObservable.pipe(
+            map(cad => {
+              const cadModel = cad.data;
+              const node = new CadNode(cadModel, this._materialManager);
+              this._materialManager.addModelMaterials(
+                cadModel.identifier,
+                cadModel.scene.maxTreeIndex,
+                cad.modelNodeAppearance
+              );
+              this._cadNodeMap.set(cadModel.identifier, node);
+              cad.callbacks.success(node);
+              return node;
+            })
+          ),
+          pointCloudObservable.pipe(
+            flatMap(
+              pointCloud => createThreeJsPointCloudNode(pointCloud.data),
+              (request, wrapper) => {
+                request.callbacks.success(wrapper);
+                return wrapper;
+              }
+            )
+          )
+        );
+      })
+    );
+
+    const cadNodeArray: CadNode[] = [];
+    this._cadObservable = this._modelObservable.pipe(
+      filter((model): model is CadNode => model instanceof CadNode),
+      scan((accumulator, cadnode) => {
+        accumulator.push(cadnode);
+        return accumulator;
+      }, cadNodeArray)
+    );
+    this._cameraSubject = this.createLoadSectorsPipeline(dataUpdatedCallback);
+  }
+
+  public addModelFromCdf(modelRevision: string | number, modelNodeAppearance?: ModelNodeAppearance): Promise<CadNode> {
+    let resolveCb: (cadNode: CadNode) => void | undefined;
+    let rejectCb: (message: string) => void | undefined;
+    const promise = new Promise<CadNode>((resolve, reject) => {
+      resolveCb = resolve;
+      rejectCb = reject;
+    });
+    this._modelSubject.next({
+      format: File3dFormat.RevealCadModel,
+      source: {
+        discriminator: 'cdf',
```

Comment:
Due to typescript I believe that you can't actually have a typo there. That is because the discriminator is a string type of said string and not a string type of string.